### ISSUE TITLE
Nouveau "placement réseau"

### ISSUE THREAD

**Issue by tcharp38** [KiwiHC16/Abeille#2585]
J'ouvre un sujet spécifique au "placement réseau", anciennement "network graph" pour les beta testeurs et discuter des evolutions prioritaires.

Le but etant d'en faire un outil clef pour comprendre les soucis de routage avec une vision permettant de localiser les equipements dans l'espace.
 L'image de fond est supposée etre le plan de l'appart/maison.

En version BETA actuellement.

Exemple
![image](https://user-images.githubusercontent.com/35221038/234862459-dcc73478-6105-4f6f-bebe-92cb422e96c5.png)

**Comment by Jake309** [comment]
Sur une maison à étage, il va surement falloir penser à intégrer un paramètre pour dire RDC ou premier étage. car les équipements seront superposés en 2D

Edit: mon réseau abeille ne remonte rien, du coup j'ai rien sur le placement réseau

**Comment by tcharp38** [comment]
Tout à fait. J'y avais pensé mais rien encore implémenté.
Je veux me focaliser sur un seul niveau (donc 1 plan) en priorité.

**Comment by tcharp38** [comment]
> Sur une maison à étage, il va surement falloir penser à intégrer un paramètre pour dire RDC ou premier étage. car les équipements seront superposés en 2D
> 
> Edit: mon réseau abeille ne remonte rien, du coup j'ai rien sur le placement réseau

Comment ca il ne remonte rien ?
Ha oui si tu as redémarré ta machine ce jour alors il faut forcer le refresh reseau via la page reseau. Il manque le bouton.
Sinon c'est fait automatiquement à minuit chaque jour.

**Comment by Jake309** [comment]
> Tout à fait. J'y avais pensé mais rien encore implémenté. Je veux me focaliser sur un seul niveau (donc 1 plan) en priorité.

oui, ca n'empeche pas. Parfois c'est moins loin de passer une dalle que de traverser la pièce ^^

ca fait beaucoup de monde : 

![image](https://user-images.githubusercontent.com/49479084/234970653-23d55eb0-e8ee-4634-adf0-5ba2df9672d3.png)

**Comment by thedrgraph** [comment]
C'est pas mal ce serait top d'avoir les trais de 3 couleur selon les LQI on pourrait vite voir les liens qui fonctionnent mal et savoir comment les renforcer avec un appareil relais. 
idéalement on devrait pouvoir masquer les liaison les moins bonne.

En tous cas même si pour mon cas je suis sur 4 niveau c'est mieux que la vue graphique des liens qui est trop brouillon quand on a un peu de monde ;)

**Comment by tcharp38** [comment]
@thedrgraph : C'est en cours
@Jake309 : Ton exemple montre qu il faut la possibilité de ne pas afficher les liens pour une meilleure visibilité lors du placement

Je vais mettre ca en priorité. A suivre...

**Comment by thedrgraph** [comment]
Autre chose pour le placement ce serait bien aussi que le nom soit écrit un peu plus gros j'ai du mal a lire en police 4 :p

**Comment by Jake309** [comment]
@tcharp38 je suis d'accord, faut pouvoir épurer. A la limite, tant qu'a fait les 3 couleurs de LQI, est ce que ca serait plus compliquer de pouvoir afficher ou non les différentes couleur ?

la police 4, ca se lit, mais c'est vrai que c'est pas le plus ergonomique. On pourrait aussi imaginer une réduction de la taille des icônes.

**Comment by tcharp38** [comment]
Je sais pas pourquoi vous parlez de police 4, la taille etant 8px. Elle est etendue à 12. Dans le futur peut etre que ce sera au choix du user.

**Comment by tcharp38** [comment]
@Jake309 @thedrgraph 
L'un de vous qui abiterait en maison pourrait m'envoyer en email privé, les plans correspondant de 2 niveaux au moins histoire d'avoir une base de travaille "multi niveau" ? Je suis en appart, 1 seul niveau.
Merci

**Comment by Jake309** [comment]
C'est dans ta boite :)

**Comment by baloo53** [comment]
Salut, j'aimerais filer un coup de main.
J'ai une maison avec un étage et un atelier à 15m de la maison. J'ai des équipements zigbee dans ceux-ci contrôlé via le plugin abeille.
Je suis actuellement sur la version bêta je pourrais vous envoyer un screen.
Pouvez vous me donner une sorte de procédure pour créer le second étage que je face sa et que je vous screen le rendu après.
Sinon je peux aussi faire d'autres manipulations que vous souhaitez. :)

**Comment by tcharp38** [comment]
Salut @baloo53 
L'idée de cet outil est d'inserer dans Abeille un plan par etage et d'y placer les equipements.
Comme ca on pourrait voir les liaisons, leur qualité de reseau et meme a terme le chemin de routage utilisé.

Je ne sais plus ou j'en étais resté mais la tu dois pouvoir deja inserer plusieurs plans mais pas de placer les equipements sur un plan choisi, pas encore.
Un plan est une image PNG d'un niveau, à creer de ton cote par le moyen qui te semble le + simple. Pour ma part j avais utilisé un truc en ligne qu'il faut que je retrouve.

**Comment by Xav-74** [comment]
Hello @tcharp38,

Je m'y suis remis ce week-end :)
J'ai un premier souci. J'upload mes 2 plans (1 par étage) mais au retour sur la page, j'ai le plan de démo qui s'affichent.
Mes fichiers sont bien dans le rep "tmp"

Second souci : si je refais un "analyser", tous mes placements disparaissent ;) et retour à l'état inital (roue)

A ta dispo

Xav

**Comment by tcharp38** [comment]
@Xav-74 
Tu as un peu de temps pour faire des tests ou pas ?
Si oui s'aligner sur la derniere beta et voila une correction pour ton plan qui ne s'affiche pas.

A decompresser dans Abeille/desktop/php
[AbeilleNetworkMap.zip](https://github.com/KiwiHC16/Abeille/files/13867076/AbeilleNetworkMap.zip)

**Comment by Xav-74** [comment]
Hello,

Dernière beta installée ! Fichier mis à jour.
Je te confirme que c'est mieux : je conserve bien mes plans cette fois-ci. Seul souci restant, lors de la seconde ouverture, la taille du plan n'est pas conservé par rapport à l'import initial. Il est bcp plus grand à l'affichage. Du coup le placement des équipements n'est pas bon ;)

Edit : en fait au départ (chargement du plan) il garde les dimensions du plan démo (1200 x 1024). Au retour sur la page, le script fait qu'il s'adapte à la taille de mon plan (1941 x 1380). Du coup je perds le placement. Ou du moins, il faut que je le refasse. A mon avis il faudrait relancer le script de dimensionnement du div au chargement du nouveau plan ;) comme ca on fait le placement au bon endroit au départ ;)

**Comment by tcharp38** [comment]
Vu que je fais des essais avec toi ca t ennuies de m'envoyer ton plan ? (tcharp38@laposte.net)
Je vais regarder des que j'ai qq mins
Merci à toi

**Comment by tcharp38** [comment]
P'tite update
[AbeilleNetworkMap.zip](https://github.com/KiwiHC16/Abeille/files/13875936/AbeilleNetworkMap.zip)

**Comment by Xav-74** [comment]
J'ai remplacé le fichier php par le nouveau.
J'ai ensuite viré mes plans du rep tmp.
Retour au plan de démo OK
Le plan est bien à côté du panel de gauche : 
![image](https://github.com/KiwiHC16/Abeille/assets/56156622/82f1febf-2796-420e-90ce-b340d215642f)

Puis je réimporte mes plans ! Premier sujet : je clique sur Edition pour finaliser, mon plan n'apparait pas. Je dois fermer mon onglet et le réouvrir pour qu'il soit pris en compte (régression par rapport à la version précédente)
Et lorsque je réouvre, le plan n'est pas à côté du panel de gauche mais en dessous ! Voulu ? Par contre les équipements ont suivi 👍 
![image](https://github.com/KiwiHC16/Abeille/assets/56156622/34775a4f-1572-45be-9718-0df24bbe09ab)

Voilà pour mes tests :)

**Comment by Xav-74** [comment]
C'est OK avec la version 240116-BETA-1 :)
👍 
Merci

**Comment by tcharp38** [comment]
Bah non.. des changements mais je ne suis pas satisfait.
En particulier si tu retrecies l ecran ou l agrandis.. il y a des choses qui ne sont plus alignées.

**Comment by tcharp38** [comment]
La derniere beta permet de placer une equipement sur un niveau choisi.

**Comment by tcharp38** [comment]
Qq retours de @MichelCHAVOT

    Bugs:
    Certains équipements s'affichent en bleu au lieu d'être en vert (non routeur). voir sur les plans RdC-2 (2 capteurs de température et 1 détecteur d'ouverture) et SousSol_2 (3 capteurs de température). Eléments cerclés en rouge.
    En mode config, quand on clique sur le carré noir à côté d'un équipement, le nom qui s'affiche est celui du dernier équipement modifié.
    
    Demandes:
    Je trouverai pratique de voir en teinte pastel les équipements qui ne sont pas sur l'étage affiché, plutôt que de les faire complètement disparaitre.

**Comment by MichelCHAVOT** [comment]
Petite remarque sur les capteurs de température qui sont considérés comme routeurs (pas tous !) alors qu'ils ne le sont pas.
d'une part, ils apparaissent en bleu au lieu d'être en vert et 
d'autre part, quand on lance une analyse réseau, ces éléments sont consultés comme des routeurs, ce qui ralentit le scan puisqu'ils ne répondent pas.

**Comment by tcharp38** [comment]
Tout à fait en ligne avec ta remarque.
Le sujet etant independant du placement réseau, j'en ai ouvert un spécifique => #2711

**Comment by tcharp38** [comment]
Qq améliorations dans la beta de demain (240630-BETA-1)
- ajout des equipements connus de Jeedom mais non visibles sur le reseau (sans vie)
- ajout d'une tete de mort pour tout equipement sans vie
- les liens sous maintenant sous l'equipement au lieu d etre dessus
- ajout d'une legende

**Comment by MichelCHAVOT** [comment]
Salut @tcharp38 
J'ai installé la dernière Béta : 2024-07-01 01:00:27
J'ai relancé le scan réseau.
Je t'envoie une image de mon placement réseau. Comme tu vois, beaucoup d'équipements avec une tête de mort, alors que tous fonctionnent très bien ? ! ?
voici les logs:
[AbeilleLogs-240701.tar.gz](https://github.com/user-attachments/files/16056072/AbeilleLogs-240701.tar.gz)

**Comment by tcharp38** [comment]
Salut
Je ne vois pas d'image mais tu m'en donnes un qui n'est rellement pas mort ? Il n'est pas en timeout par hasard ?

**Comment by tcharp38** [comment]
Ok l'analyse reseau dit qu'elle n'a pas eu de reponse (time out) sur 5 routeurs.
C est ce que tu vois ? 5 tetes de mort ?

**Comment by MichelCHAVOT** [comment]
Oups ! Voici l'image : 
![image](https://github.com/KiwiHC16/Abeille/assets/36966985/e356c91c-59d3-4ed4-855c-b219b4804192)
Ci-joint, aussi un extrait de la timeline ou tu pourras voir que tous mes capteurs de température renvoient des mesures très régulièrement.
[timeline.log](https://github.com/user-attachments/files/16058380/timeline.log)

**Comment by tcharp38** [comment]
Ok j'ai compris ..au moins une piste.

Durant l'analyse, on interroge tous les routeurs pour avoir la table dite "LQI" (liste des liens)
Hors 5 ne repondent pas à ce moment.
Du coup tous les enfants qui en decoulent n'ont aucun lien (ex Temp Parents) et ils sont ajoutés en "dead" car non listés dans le réseau.

La question du coup est pourquoi ces 5 time out. (AbeilleLqi.log)

**Comment by tcharp38** [comment]
A faire (pour moi) .. telecommandes virtuelles a ne pas marquer "sans vie"

**Comment by MichelCHAVOT** [comment]
Bonjour.
J'ai recherché tous les Time-Out dans le fichier AbeilleLqi.log que je t'ai envoyé. Les voici : 
![image](https://github.com/KiwiHC16/Abeille/assets/36966985/d83924ef-92a7-4f6b-b758-6edf7e01e00a)
J'ai contrôlé à quelle valeur était réglé ce Time-Out sur chacun de ces équipements dans abeille.
Ils sont tous à 60 minutes. exemple : 
![image](https://github.com/KiwiHC16/Abeille/assets/36966985/e83a5071-0c4c-4890-adb4-2d93f6f62d33)
J'ai déjà remarqué que pendant le scan réseau, certains équipements répondent tout de suite, et pour d'autres çà traine bien 5 secondes. (mais pas 60 minutes !)
Peut-être peux-tu de ton coté augmenter un peu le Time-Out du scan, ou bien réinterroger 1 ou 2 fois l'équipement qui est en Time-Out.
Pour info, j'ai déjà constaté que mon système rame (à mon gout). il faut parfois plusieurs secondes quand j'envoie une commande sur un équipement avant que celui-ci ne réagisse.

**Comment by tcharp38** [comment]
Tiens voila une nouvelle version.
J'autorise 3 reessais si time out sur un equipement.
[AbeilleLQI.zip](https://github.com/user-attachments/files/16068187/AbeilleLQI.zip)

Rappel: pour Abeille/core/php
et file moi le package de logs STP

**Comment by MichelCHAVOT** [comment]
Voici les résultats du test : 
![image](https://github.com/KiwiHC16/Abeille/assets/36966985/02d2b4db-460a-4f49-8109-11746257c55f)
[AbeilleLogs-240702 (1).tar.gz](https://github.com/user-attachments/files/16071943/AbeilleLogs-240702.1.tar.gz)

**Comment by tcharp38** [comment]
Et avec celui la ?
[AbeilleLQI.zip](https://github.com/user-attachments/files/16073775/AbeilleLQI.zip)

**Comment by MichelCHAVOT** [comment]
Toujours des équipements qui ne répondent pas, pourtant, si tu regardes le time-time des capteurs de température, ils sont tous récents (à comparer avec l'hehure du scan réseau) : 
![image](https://github.com/KiwiHC16/Abeille/assets/36966985/d3a1b39e-1658-4a48-9380-4ea653647dbf)
![image](https://github.com/KiwiHC16/Abeille/assets/36966985/1d1c229c-e988-4e80-ae24-1a3569b1474f)
[AbeilleLogs-240703.tar.gz](https://github.com/user-attachments/files/16080677/AbeilleLogs-240703.tar.gz)

**Comment by tcharp38** [comment]
Durant l'interrogation j'en vois 3 qui ne repondent pas
- 'VMC Appart'
- 'Guirlandes extérieures'
- 'Prise TV'

mais cette fois l'interrogation refait une passe et 2 finissent par repondre
- 'VMC Appart'
- 'Guirlandes extérieures'
Donc ces 2 la ne doivent pas avoir une tete de mort

Par contre le 3iem... 'Prise TV'.. ne repond pas malgré 3 requetes.
Lui doit avoir une tete de mort.
Correct ?

**Comment by tcharp38** [comment]
Tiens une update pour Abeille/desktop/php
[AbeilleNetworkMap.zip](https://github.com/user-attachments/files/16082378/AbeilleNetworkMap.zip)

Ca va ne plus afficher les telecommandes virtuelles.

**Comment by MichelCHAVOT** [comment]
Les têtes de mort s'affichent sur les End Device, pas sur les routeurs. Prise TV est un routeur. même s'il ne répond pas, il n'a pas de tête de mort. Sur l'image Réseau, la prise TV est tout en bas, quasiment au centre, la bleue la plus à gauche.
Les télécommandes virtuelles ne s'affichent plus.
![image](https://github.com/KiwiHC16/Abeille/assets/36966985/7ffed625-84a5-4b18-b4fd-79c28cbbb9d6)
[AbeilleLogs-240703.tar.gz](https://github.com/user-attachments/files/16085959/AbeilleLogs-240703.tar.gz)
Comme tu peux le voir sur mon virtuel de contrôle, les time-time sont tous récents : 
![image](https://github.com/KiwiHC16/Abeille/assets/36966985/dc0fea6c-59d9-4755-bf59-2d87750627d5)

**Comment by tcharp38** [comment]
Est ce que tu confirmes que tous les devices qui sont marqués "dead" sont
- des end devices
- tous sont dessinés sans lien
- il semble qu'aucun router ne dise qu'il les a dans son sillage

**Comment by MichelCHAVOT** [comment]
Tous les devices qui sont marqués "dead" sont des end devices.
Tous les devices qui sont marqués "dead" sont dessinés sans lien.
il semble qu'aucun router ne dise qu'il les a dans son sillage, et pourtant tous mes devices fonctionne ! çà passe bien quelque part dans le réseau !

**Comment by tcharp38** [comment]
Ok donc au final ils sont marqués "dead" car pas dans la liste de voisinage.
Cette liste n'est donc pas la bonne à utiliser car incomplete.

**Comment by MichelCHAVOT** [comment]
Bon courage !

**Comment by tcharp38** [comment]
Pour la beta de demain (240704-BETA-1).. tout equipement connu de Jeedom mais pas trouvé sur le reseau ne sera plus marqué "mort" jusqu'a trouver mieux.
